### ISSUE TITLE
[BUG] cloudbeat crash once etcd folder changed group and user

### ISSUE THREAD

**Issue by CohenIdo** [elastic/cloudbeat#118]
**Describe the bug**
After changing `/var/lib/etcd` user and/or group to `etcd` cloudbeat carsh.

**Preconditions**
Environment state and/or platform specific configurations.

**To Reproduce**
Write the exact actions one should perform in order to reproduce the bug.
1. initiate cloudbeat
2. ssh to the cluster
3. run:
     ```
    groupadd etcd
    useradd -g etcd etcd
    chown etcd:etcd /var/lib/etcd/
   ```
4.  wait 5-10 seconds and check for cloudbeat logs

**Expected behavior**
get findings as usual 

**Screenshots**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x204cee4]

goroutine 150 [running]:
github.com/elastic/cloudbeat/resources/fetchers.FromFileInfo({0x2f1b060, 0x40007cd200}, {0x40001460c0, 0x14})
	/Users/idocohen/Documents/es/cloudbeat/resources/fetchers/file_system_fetcher.go:113 +0x1c4
github.com/elastic/cloudbeat/resources/fetchers.(*FileSystemFetcher).fetchSystemResource(0x40007d0ea0, {0x40001460c0, 0x14})
	/Users/idocohen/Documents/es/cloudbeat/resources/fetchers/file_system_fetcher.go:84 +0x150
github.com/elastic/cloudbeat/resources/fetchers.(*FileSystemFetcher).Fetch(0x40007d0ea0, {0x2efa680, 0x400079abc0})
	/Users/idocohen/Documents/es/cloudbeat/resources/fetchers/file_system_fetcher.go:66 +0x2a8
github.com/elastic/cloudbeat/resources/manager.(*fetchersRegistry).Run(0x40002fa7a0, {0x2efa680, 0x400079abc0}, {0x40004221e0, 0xb})
	/Users/idocohen/Documents/es/cloudbeat/resources/manager/registry.go:97 +0xc8
github.com/elastic/cloudbeat/resources/manager.(*Data).fetchWorker(0x40007ee2a0, {0x2efa680, 0x400079abc0}, 0x4000157440, {0x40004221e0, 0xb})
	/Users/idocohen/Documents/es/cloudbeat/resources/manager/data.go:97 +0xbc
github.com/elastic/cloudbeat/resources/manager.(*Data).Run.func1(0x40007b05f0, 0x40007ee2a0, {0x2efa680, 0x400079abc0}, 0x4000157440, {0x40004221e0, 0xb})
	/Users/idocohen/Documents/es/cloudbeat/resources/manager/data.go:74 +0x78
created by github.com/elastic/cloudbeat/resources/manager.(*Data).Run
	/Users/idocohen/Documents/es/cloudbeat/resources/manager/data.go:72 +0x198
```

**code reference**
`cloudbeat/resources/fetchers/file_system_fetcher.go:84 `

```
uid := stat.Uid
	gid := stat.Gid
	u := strconv.FormatUint(uint64(uid), 10)
	g := strconv.FormatUint(uint64(gid), 10)
	usr, _ := user.LookupId(u)
	group, _ := user.LookupGroupId(g)
	mod := strconv.FormatUint(uint64(info.Mode().Perm()), 8)
	inode := strconv.FormatUint(uint64(stat.Ino), 10)

	data := FileSystemResource{
		FileName: info.Name(),
		FileMode: mod,
		Uid:      usr.Name,
		Gid:      group.Name,
		Path:     path,
		Inode:    inode,
	}

```

**Desktop (please complete the following information):**
 - OS: 
 - Browser: 
 - Kibana Version: 
 - Endpoint Version:
 - Other Version: 

**Additional context**
cloudbeat commit sha: `03423461f38b60d89925bb810f40f997a4143a4c`

**Comment by uri-weisman** [comment]
**Issue:**

Cloudbeat is using the [`os/user`](https://github.com/elastic/cloudbeat/blob/main/resources/fetchers/file_system_fetcher.go#L111-L112) native go pkg for obtaining file's permissions and the pkg uses the [`/etc/passwd`](https://github.com/golang/go/blob/master/src/os/user/lookup_unix.go#L19) file to do so.

In the described scenario, we change the file's permissions from the host and expect to gather data about this file from inside the container - but the host and container files are not aligned (`/etc/passwd` & `/etc/group`).

A suggested solution is to add two new volumes and mount `/etc/passwd` & `/etc/group`, both files are already mounted to a different path.

@eyalkraft, @jeniawhite, @DaveSys911  - would appreciate your opinion.

**Comment by eyalkraft** [comment]
Nice update! Following our daily please update if you can utilize some other functions of `os/user` to fix this bug.

**Comment by uri-weisman** [comment]
> Nice update! Following our daily please update if you can utilize some other functions of `os/user` to fix this bug.

Unfortunately, it seems to be impossible.

I can think of two options to solve this issue:
1. Implement our own solution - getting a user's information from a different file, e.g. `"/hostfs/var/lib/etcd"` which is already mounted.
2. mount  `-v /etc/passwd:/etc/passwd:ro`
`-v /etc/group:/etc/group:ro`
That could be made to work, but it does have security implications.

**Comment by eyalkraft** [comment]
Hey @uri-weisman, I think option 1 will be our way to go.
Can you already estimate the work necessary for the fix?

**Comment by uri-weisman** [comment]
Additional bugs were found:
1. Currently we use the `user.name` as the `Uid` [(link to source code),](https://github.com/elastic/cloudbeat/blob/main/resources/fetchers/file_system_fetcher.go#L119).
The field is a human-readable name generated by adding a comment upon user creation (`useradd -c "System administrator" root`).
After consulting with @eyalkraft we decided to use the `user.username` field which is the actual user - will be fixed together with this bug.

For example, we should use **list** instead of **Mailing List Manager**.
![Screen Shot 2022-05-17 at 12 53 18](https://user-images.githubusercontent.com/68195305/168784253-1b4e1d0b-203e-408c-a1e4-e363d844baa6.png)


2. We assign `user.username`, `group.name` to `Uid` and `Gid` respectively, fields names don't represent the data they store, I'll open a ticket to fix it after the 8.3 release as it involves with changes we don't want to apply at the moment.
